Taxi/Taxi/Presenter/TaxiPartyList/FilterBar.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  FilterBar.swift
+//  Taxi
+//
+//  Created by JongHo Park on 2022/08/07.
+//
+
+import SwiftUI
+
+// MARK: - Filter bar
+struct FilterBar: View {
+    private let onFilterSelected: (TaxiPartyFilter) -> Void
+
+    private let filters: [TaxiPartyFilter] = [EmptyFilter(), PohangFilter(), PostechFilter()]
+    @Binding private var showCalendarModal: Bool
+    @State private var selectedIndex: Int = 0
+    @State private var cgRects: [CGRect] = Array(repeating: .zero, count: 3)
+
+    init(_ showCalendarModal: Binding<Bool>, onFilterSelected: @escaping (TaxiPartyFilter) -> Void) {
+        self._showCalendarModal = showCalendarModal
+        self.onFilterSelected = onFilterSelected
+    }
+    var body: some View {
+        HStack(alignment: .firstTextBaseline) {
+            Spacer().frame(width: 16)
+            ForEach(0..<3) { index in
+                Button {
+                    withAnimation(.interactiveSpring()) {
+                        selectedIndex = index
+                        onFilterSelected(filters[index])
+                    }
+                } label: {
+                    Text(filters[index].filterName)
+                }
+                .padding(EdgeInsets(top: 16, leading: 8, bottom: 8, trailing: 8))
+                .background(GeometryReader { proxy in
+                    Color.clear.onAppear {
+                        cgRects[index] = proxy.frame(in: .named("FilterSection"))
+                    }
+                })
+            }
+            Spacer()
+            dateSelector
+            Spacer().frame(width: 16)
+        }
+        .foregroundColor(.customBlack)
+        .overlay(alignment: .bottom) {
+            Rectangle()
+                .fill(Color.lightGray)
+                .frame(height: 1)
+        }
+        .overlay {
+            Rectangle()
+                .fill(Color.black)
+                .frame(width: cgRects[selectedIndex].width, height: 2)
+                .position(x: cgRects[selectedIndex].midX, y: cgRects[selectedIndex].maxY - 1)
+
+        }
```

Comment:
현재 검은색 필터 라인의 위치를 `.position(x: cgRects[selectedIndex].midX, y: cgRects[selectedIndex].maxY - 1)`로 y축 값도 지정해주고 있는데, 검은색 필터를 그려주는 .overlay 코드를 회색 라인을 수식하게 해주면 알아서 중앙 정렬이 되어 y축 위치를 지정해주는 코드를 없애줄 수 있을 것 같은데 어떻게 생각하시나요?

```suggestion
        .overlay(alignment: .bottom) {
            // lightGray divider line...
                .overlay {
                    // black filter line...
                        .position(x: cgRects[selectedIndex].midX)
                }
        }
```